### PR TITLE
[#6] [Android] [UI] As a user, I can log in to the app

### LINKED ISSUE #6: [Android] [UI] As a user, I can log in to the app
## Why

When the user comes to the Login screen, they can log in using their email and password.
 
## Acceptance Criteria

- Show the Login screen with the logo on top from the Splash screen
- Show the `email` and `password` inputs, the email should have email input format and password in password input format.
- Show the `Forgot?` link on the right of Password input.
- Show the `Log in` button on the bottom of login form.

Design: https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=31%3A195&scaling=scale-down

Sign in

### PR DESCRIPTION
**Pull request by luongvo**
- Close #6

## What happened 👀

- Show the Login screen with the logo on top of the screen
- Show the `email` and `password` inputs, the email should have the email input format and password in the password input format.
- Show the `Forgot?` link on the right of the Password input.
- Show the `Log In` button at the bottom of the login form.

## Insight 📝

- I created a `DimmedImageBackground` composable to build a blur + gradient background effect.
- I created a `CustomTextField` to build the custom rounded text input.
- I created a `CustomButton` to build the custom rounded button.

## Proof Of Work 📹

https://user-images.githubusercontent.com/16315358/208029026-ec938988-7776-4aa9-9e07-721c43033354.mp4

### CODE REVIEW

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/CustomButton.kt
Comment (suho) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/CustomButton.kt]: It's better to avoid words with unclear meanings. I prefer to use `PrimaryButton` instead. Furthermore, if we have to customize another button (e.g., `SecondaryButton`), using `Custom` will not be suitable for expanding.

We have [this convention](https://nimblehq.co/compass/development/code-conventions/swift/#naming-classes-should-be-straightforward) on Swift page, but I think it's ok to mention here :D

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt
Comment (suho) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt:19]: We might use this with an image URL from API, so we might need to improve this.

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt
Comment (suho) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt:37]: Should we have `onLoginButtonClick` for consistency?

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt
Comment (suho) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt]: @luongvo Just curious, why don't we just use `.offset(0.dp, (-299).dp)`?

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/theme/AppTypography.kt
Comment (suho) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/theme/AppTypography.kt]: `h4` size should bigger than `h5` size IMO
```suggestion
    h4 = TextStyle(
        fontSize = 34.sp,
        fontWeight = FontWeight.Bold
    ),
    h5 = TextStyle(
        fontSize = 26.sp,
        fontWeight = FontWeight.Bold
    ),
```

File: detekt-config.yml
Comment (suho) [inline detekt-config.yml:191]: 👍

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt:19]: Yes, but not this time @suho 😛

**Review by blyscuit — COMMENTED** [review]
LGMT 👍

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/CustomButton.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/CustomButton.kt]: @suho Good point, fixed in https://github.com/luongvo/kmm-survey/pull/48/commits/2bf5858983ff990d311a04c1e83df985919b6207

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt:37]: Added in https://github.com/luongvo/kmm-survey/pull/48/commits/2bf5858983ff990d311a04c1e83df985919b6207

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginScreen.kt]: Fixed in https://github.com/luongvo/kmm-survey/pull/48/commits/2bf5858983ff990d311a04c1e83df985919b6207

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/theme/AppTypography.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/theme/AppTypography.kt]: Fixed in https://github.com/luongvo/kmm-survey/pull/48/commits/2bf5858983ff990d311a04c1e83df985919b6207

**Review by suho — APPROVED** [review]

**Comment by luongvo** [comment]
@blyscuit Can I get your official approval? 😄

**Review by blyscuit — APPROVED** [review]
Would be good to wait for more Android's chapter approval.

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt
Comment (doannimble) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt:28]: @luongvo Can we ignore this property? 🤔

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt
Comment (doannimble) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt:25]: @luongvo I think this property should be `optional`. 🤔

**Review by doannimble — COMMENTED** [review]
rest lgtm

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt:28]: @doannimble Can't, it's a required field

image

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt:25]: @doannimble it's a required field from the original TextField so I would like to follow that constraint. Actually we need to catch all input events to get its value 😄 

image

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt
Comment (doannimble) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/DimmedImageBackground.kt:28]: @luongvo Oh, how funny it is, nullable but required 😄 . Agree with the current approach 👌

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt
Comment (doannimble) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt:25]: @luongvo Thanks for the clarification. In some cases, we only need to get the final text without checking the text change.
We're defining the common, so in my humble opinion, forcing it is not a good idea. But anyway, we should keep it as the original TextField is required.

**Review by doannimble — APPROVED** [review]
All clear and lgtm 🚀

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/common/PrimaryTextField.kt:25]: @doannimble Note that, Jetpack Compose is far from the XML method. I'm wondering how can we get the input value without having `onValueChange` 😄 
- https://developer.android.com/jetpack/compose/text#enter-modify-text 
- https://www.tutorialkart.com/android-jetpack-compose/textfield-read-value/

**Pull request merged**
